internal/clients/ionoscloud.go

```diff
@@ -126,3 +126,19 @@ func ConnectForCRD(ctx context.Context, mg resource.Managed, client kubeclient.C
 	}
 	return svc, nil
 }
+
+// DataCenterElement is an ionos cloud API object with metadata
+type DataCenterElement interface {
```

Comment:
I agree this is useful, and especially the method below, but I suggest to rename this to something without `DataCenter` as it can be confusing. This should be used for all services, not only related to `DataCenter`. I think `Resource` or `StateElement` is more suggestive.

Reply:
Called it `APIResource`. This allows us to extend the interface with other methods every APIResource of the IONOS API has.